ecs-cli/modules/clients/aws/cloudformation/template.go

```diff
@@ -155,7 +155,7 @@ var template = `
       "Type": "String",
       "Description": "Optional - VPC Id of existing VPC. Leave blank to have a new VPC created",
       "Default": "",
-      "AllowedPattern": "^(?:vpc-[0-9a-f]{8}|)$",
+      "AllowedPattern": "^(?:vpc-[0-9a-f]{8,17}|)$",
```

Comment:
I believe this would allow ids between 8 and 17 characters, and if I'm understanding the new format correctly, the new longer IDs are exactly 17 characters.  Could you modify your regex to only support 8 or 17 characters or an empty string? (I tried `^(vpc-[0-9a-f]{8}|vpc-[0-9]{17}|)$`, which seems to work but I believe [Cloudformation](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/cfn-regexes.html) uses the Java regex library so there may be other ways as well).

Reply:
I have changed the pattern to ^(?:vpc-[0-9a-f]{8}|vpc-[0-9a-f]{17}|)$ and test the regex at http://www.regexplanet.com/advanced/java/index.html with input of:
vpc-06da6846b38178d68
vpc-2605db41
vpc-2605db412

Only the first 2 match. The last one does not match

Reply:
Great, thanks!